packages/headless/src/features/recent-queries/recent-queries-actions.ts

```diff
@@ -0,0 +1,36 @@
+import {createAction} from '@reduxjs/toolkit';
+import {ArrayValue, NumberValue, StringValue} from '@coveo/bueno';
+import {validatePayload} from '../../utils/validate-payload';
+
+export interface RegisterRecentQueriesCreatorPayload {
+  /**
+   * The recent queries made by the user prior to instantiating the controller.
+   */
+  queries: string[];
+  /**
+   * The maximum number of queries to retain in the list.
+   */
+  maxQueries: number;
+}
+
+const registerRecentQueriesPayloadDefinition = {
+  queries: new ArrayValue({required: true, each: new StringValue()}),
```

Comment:
Probably you would want to make sure that each `StringValue` is non empty. Saving "empty query" in the state should not be relevant.